workflows/storage/unarchive.yaml

```
@@ -0,0 +1,213 @@
+        value: 'linz-hydrographic-archive'
+        enum:
+          - 'linz-topographic-archive'
+          - 'linz-hydrographic-archive'
+      - name: archive_directory
+        description: 'The directory in the archive bucket where the data to unarchive is stored. Does not include the bucket name. Example: "aerialsurveys/Invercargill2022_Pgrm3016/".'
+        value: ''
+      - name: consumer
+        description: 'The consumer for who the data is being unarchived. The unarchive data will be copied to the consumer shared location.'
+        value: 'linz'
+        enum:
+          - 'linz'
+      - name: urgent
+        description: 'Whether the unarchive request is urgent (with a cost impact). If not urgent ("BULK" retrieval tier), the restore process can take up to 48 hours. If urgent ("STANDARD" retrieval tier), it will take up to 12 hours. Defaults to false.'
+        value: 'false'
+        enum:
+          - 'true'
+          - 'false'
+      - name: group
+        value: '1000'
+      - name: group_size
+        value: '100Gi'
+  templates:
+    - name: main
+      dag:
+        tasks:
+          - name: create-manifest
+            templateRef:
+              name: tpl-create-manifest
+              template: main
+            arguments:
+              parameters:
+                - name: source
+                  value: 's3://{{workflow.parameters.archive_bucket}}/{{=sprig.trimPrefix("/",workflow.parameters.archive_directory)}}'
+                - name: target
+                  value: 's3://{{= workflow.parameters.archive_bucket == "linz-topographic-archive" ? "linz-topographic-shared" : workflow.parameters.archive_bucket == "linz-hydrographic-archive" ? "linz-hydrographic-shared" : "unknown-bucket" }}/{{workflow.parameters.consumer}}/{{=sprig.trimPrefix("/",workflow.parameters.archive_directory)}}'
+                - name: include
+                  value: '.*'
+                - name: group
+                  value: '{{workflow.parameters.group}}'
+                - name: group_size
+                  value: '{{workflow.parameters.group_size}}'
+                - name: aws_role_config_path
+                  value: 's3://linz-bucket-config/config-write.topographic.json, s3://linz-bucket-config/config-write.hydrographic.json'
+          - name: get-location
+            templateRef:
+              name: tpl-get-location
+              template: main
+          - name: restore
+            template: restore
+            arguments:
+              parameters:
+                - name: archive_bucket
+                  value: '{{workflow.parameters.archive_bucket}}'
+                - name: archive_directory
+                  value: '{{=sprig.trimSuffix("/",sprig.trimPrefix("/",workflow.parameters.archive_directory))}}'
+                - name: archive_path
+                  value: '{{workflow.parameters.archive_bucket}}/{{=sprig.trimSuffix("/",sprig.trimPrefix("/",workflow.parameters.archive_directory))}}'
```

```suggestion
                  value: '{{workflow.parameters.archive_bucket}}/{{=sprig.trimSuffix("/",sprig.trimPrefix("/",sprig.trim(workflow.parameters.archive_directory)))}}'
```
in case of white space from copy & paste